vnext/Microsoft.ReactNative/IReactContext.idl

```diff
@@ -10,5 +10,8 @@ namespace Microsoft.ReactNative {
     void DispatchEvent(Windows.UI.Xaml.FrameworkElement view, String eventName, JSValueArgWriter eventDataArgWriter);
     void CallJSFunction(String moduleName, String methodName, JSValueArgWriter paramsArgWriter);
     void EmitJSEvent(String eventEmitterName, String eventName, JSValueArgWriter paramsArgWriter);
+
+    // Data attached to the instance through ReactInstanceSettings
+    Object UserData { get; };
```

Comment:
>Object UserData { get; }; [](start = 4, length = 25)

The IReactContext is shared across multiple native modules and we must be able to associate multiple values.
It must not be a single value like this.
It may be good for a quick test, but it is not usable in any serious scenario.

Please do not submit this PR.

Reply:
This PR was just me playing around. No intent to submit (Hence the single "[Not for check in]" title).

II'm not sure how much of the thread you were following, but I was initially thinking there was some niceness around a single object that is explicitly modeled. I.e. the user puts a struct there with the state they need.

That ends up being really nice in simple cases, but does break down for cross language and community modules, so I don't think it would work.

Reply:
Do we have multiple threads? I did not see any discussion about having a single object to be a good thing. I thought we always talked about a property bag that we had in Office implementation (https://github.com/microsoft/react-native-windows/blob/master/vnext/Microsoft.ReactNative/ReactHost/PropertyBag.h) and discussed questions around how to do it right for the ABI-safe cross-language world.